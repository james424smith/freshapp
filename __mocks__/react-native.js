import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
