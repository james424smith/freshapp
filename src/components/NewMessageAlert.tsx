import React, { useEffect, useRef } from "react";
import { useSelector } from "react-redux";
import DropdownAlert from "react-native-dropdownalert";
import { IRootReducerType } from "../redux/reducers";
import { ChatPushNotificationToSet } from "../interfaces";

const NewMessageAlert = () => {
  const dropdownRef = useRef<DropdownAlert | null>(null);
  const notificationToShow = useSelector<
    IRootReducerType,
    ChatPushNotificationToSet | undefined
  >(({ websocketReducer }) => websocketReducer.notificationToShow);

  useEffect(() => {
    if (notificationToShow && dropdownRef.current) {
      const body = notificationToShow.notification.notificationBody;
      const splitter = body.lastIndexOf("says:") + 6;
      dropdownRef.current.alertWithType(
        "info",
        body.substr(0, splitter),
        body.substr(splitter)
      );
    }
  }, [notificationToShow]);

  return (
    <DropdownAlert ref={dropdownRef} closeInterval={6000} zIndex={1000000} />
  );
};

export default NewMessageAlert;
