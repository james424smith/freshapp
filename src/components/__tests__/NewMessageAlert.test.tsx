import React from "react";
import NewMessageAlert from "../NewMessageAlert";
import { render, cleanup } from "@testing-library/react-native";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

describe("NewMessageAlert snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  beforeEach(() => {
    jest.useFakeTimers();
  });

  it("should render correct ui when notificationToShow is defined", () => {
    const mockedState = {
      websocketReducer: {
        notificationToShow: {
          notification: {
            notificationBody: "Giorgos says: Hello there",
          },
        },
      },
    };
    const mockedStore = configureMockStore();

    const wrapper = render(
      <Provider store={mockedStore(mockedState)}>
        <NewMessageAlert />
      </Provider>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it("should render correct ui when notificationToShow is undefined", () => {
    const mockedState = {
      websocketReducer: {
        notificationToShow: undefined,
      },
    };
    const mockedStore = configureMockStore();

    const wrapper = render(
      <Provider store={mockedStore(mockedState)}>
        <NewMessageAlert />
      </Provider>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
