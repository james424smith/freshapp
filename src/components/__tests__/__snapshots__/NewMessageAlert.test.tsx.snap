// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMessageAlert snapshot test should render correct ui when notificationToShow is defined 1`] = `
<View
  forwardedRef={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
      "zIndex": 1000000,
    }
  }
>
  <View
    accessible={false}
    focusable={true}
    onClick={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
          },
          Object {
            "backgroundColor": "#2B73B6",
          },
        ]
      }
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={Object {}}
          style={
            Object {
              "alignSelf": "center",
              "height": 36,
              "padding": 8,
              "width": 36,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "padding": 8,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "left",
              }
            }
            text="Giorgos says: "
          >
            Giorgos says: 
          </Text>
          <Text
            numberOfLines={3}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 14,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
            text="Hello there"
          >
            Hello there
          </Text>
        </View>
      </RCTSafeAreaView>
    </View>
  </View>
</View>
`;

exports[`NewMessageAlert snapshot test should render correct ui when notificationToShow is undefined 1`] = `null`;
